.github/workflows/pages.yml

```diff
@@ -56,10 +56,70 @@ jobs:
         path: doxygen.tar
         retention-days: 1
 
-  pages:
+  check-doc:
     runs-on: ubuntu-latest
     needs: [docker-build, id_repo]
     steps:
+    - name: Free Disk Space (Ubuntu)
+      uses: jlumbroso/free-disk-space@v1.3.1
+      with: { tool-cache: true, large-packages: false }
+    - name: Checkout built branch
+      uses: actions/checkout@v4
+      with:
+        submodules: recursive
+        fetch-depth: 0
+    - name: Collect image artifact
+      if: needs.id_repo.outputs.in_base_repo == 'false'
+      uses: actions/download-artifact@v3
+      with: { name: 'doxygen-artifact' }
+    - name: Load image artifact into docker
+      if: needs.id_repo.outputs.in_base_repo == 'false'
+      run: |
+        docker load < doxygen.tar
+        rm doxygen.tar
+    - name: Generate documentation
+      run: |
+        cat<<-EOF > run.sh
+          set -xe
+          git config --global --add safe.directory '*'
+          cmake \
+              -DDDC_BUILD_DOCUMENTATION=ON \
+              -DCMAKE_CXX_STANDARD=20 \
+              -DDOXYGEN_EXTRACT_ALL=NO \
+              -DDOXYGEN_WARN_IF_UNDOCUMENTED=YES \
+              -DDOXYGEN_WARN_IF_DOC_ERROR=YES \
+              -DDOXYGEN_WARN_IF_INCOMPLETE_DOC=YES \
+              -DDOXYGEN_WARN_NO_PARAMDOC=YES \
+              -DDOXYGEN_WARN_IF_UNDOC_ENUM_VAL=YES \
+              -DDOXYGEN_WARN_AS_ERROR=NO \
```

Comment:
Now I understand why you changed the parameters as you change them back here. But I am still not sure how well this works. Did you test to ensure that the errors are correctly raised for the .hpp files? 
There are some parameters mentioned in the CMake with the same configuration as here. Did you simply include all `DOXYGEN_WARN_XXX` flags for completeness?

Reply:
Yes I copy-pasted the content, in case I decide to change the default later on.

I have just commit a missing parameter documentation in a function. Let us see if it works

Reply:
Errors are now raised: https://github.com/CExA-project/ddc/actions/runs/8663814850/job/23758836602?pr=403